packages/manager/modules/product-offers/src/workflows/product-offers-workflow.class.js

```diff
@@ -79,6 +80,35 @@ export default class Workflow {
     return this.pricings && this.pricings.length === 1;
   }
 
+  /**
+   * Determines if the offer has a pay-as-you-go option
+   * @return {boolean}
+   */
+  hasComsuptionOption() {
+    const addonFamilies = get(
+      this.catalog[this.catalogItemTypeName].find(
+        ({ planCode }) => planCode === this.getPlanCode(),
+      ),
+      'addonFamilies',
+    );
+    if (!addonFamilies) {
+      return false;
+    }
+    const addonNew = addonFamilies.find((addon) => addon.name === 'NEW');
+    if (!addonNew) {
+      return false;
+    }
+    const addons = addonNew.addons.reduce((result, addonName) => {
+      result.push(
+        this.catalog.addons.find((addon) => addon.planCode === addonName),
+      );
+      return result;
+    }, []);
+    return !!addons.find((addon) =>
+      addon.pricings.find((pricing) => pricing.type === 'consumption'),
+    );
```

Comment:
```suggestion
    return addons.has(({ pricings }) =>
      pricings.has(({ type }) => type === 'consumption'),
    );
```
Also, `consumption` should be stored in https://github.com/ovh/manager/blob/master/packages/manager/modules/product-offers/src/pricing/pricing.constants.js